packages/node_modules/@ciscospark/internal-plugin-metrics/src/metrics.js

```diff
@@ -99,6 +114,107 @@ const Metrics = SparkPlugin.extend({
       eventPayload: payload
     };
     return this.callDiagnosticEventsBatcher.request(event);
+  },
+
+  storeMetric(id, metricProps, metricType) {
+    const tuple = {metric: metricProps, type: metricType};
+    return this.metricMap.set(id, tuple);
+  },
+
+  getMetrics() {
+    return this.metricMap;
+  },
+
+  getMetric(id) {
+    return this.metricMap.get(id);
+  },
+
+  deleteMetric(id) {
+    return this.metricMap.delete(id);
+  },
+
+  registerMetric(eventName, id, props, type) {
+    if (!this.getMetric(id)) {
+      // hash name and props into table/object
+      const payload = this.createPayload(eventName, props);
+      return this.storeMetric(id, payload, type);
+    }
+    return false;
+  },
+
+  /**
+  * Registers a metric to be used to later update or call finish method
+  * @param {string} eventName name of the metric
+  * @param {string} id unique identifier
+  * @param {Object} props to populate the metric with
+  * @returns {bool} true if metric registered, false otherwise
+  */
+  clientRegisterMetric(eventName, id, props) {
+    return this.registerMetric(eventName, id, props, this.CLIENT_METRIC);
+  },
+
+  /**
+  * Registers a metric to be used to later update or call finish method
+  * @param {string} eventName name of the metric
+  * @param {string} id unique identifier
+  * @param {Object} props to populate the metric with
+  * @returns {bool} true if metric registered, false otherwise
+  */
+  sdkRegisterMetric(eventName, id, props) {
```

Comment:
Let's call this `registerMetric()`. 

If there's already a client metric, this should behave like `updateRegisteredMetric()`, right?

Reply:
This is an abstraction of the underlying implementation so we can change the functionality of the method later. It is not strictly necessary, but prevents us from having to use the same preprocessing for both client and SDK metrics.

Is just a simple wrapper.

Reply:
yea, understood. i was thinking it made more sense for the method called by the sdk to be a pretty generic name, and then the method called by consumers would be obviously for consumers.

Reply:
Right, right; got it. Since we're thinking about this in the context of the SDK, the generic method should be used by the SDK, whereas the explicitly differentiated one would be for external users.

I can change the names of the SDK ones, sure.